tests/e2e/produce_consume.go

```diff
@@ -0,0 +1,88 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// requireDeployingKcatPod deploys kcat pod form a template and checks the pod readiness
+func requireDeployingKcatPod(kubectlOptions k8s.KubectlOptions, podName string) {
+	It("Deploying Kcat Pod", func() {
+		applyK8sResourceFromTemplate(kubectlOptions,
+			kcatPodTemplate,
+			map[string]interface{}{
+				"Name":      kcatPodName,
+				"Namespace": kubectlOptions.Namespace,
+			},
+		)
+		err := waitK8sResourceCondition(kubectlOptions, "pods", "condition=Ready", defaultPodReadinessWaitTime, "", kcatPodName)
+		Expect(err).ShouldNot(HaveOccurred())
+	})
+
```

Comment:
**Required**: We **MUST** check potential errors here. (also bit of a formatting advice)

```suggestion
		err := applyK8sResourceFromTemplate(kubectlOptions,
			kcatPodTemplate,
			map[string]interface{}{
				"Name":      kcatPodName,
				"Namespace": kubectlOptions.Namespace,
			},
		)
		
		Expect(err).NotTo(HaveOccurred())
		
		err := waitK8sResourceCondition(kubectlOptions, "pods", "condition=Ready", defaultPodReadinessWaitTime, "", kcatPodName)
		
		Expect(err).ShouldNot(HaveOccurred())
	})
```

Reply:
fixed: [3efc5f9](https://github.com/banzaicloud/koperator/pull/987/commits/3efc5f92e73093a477784196d6bcf64ffcc0a417)